### PR TITLE
Add unit tests for ReadStringContext function

### PR DESCRIPTION
**Pull request by PabloHiro**
## Summary
Adds comprehensive unit tests for `pkg/utils/readstring_context.go` to improve test coverage and code quality.

## Test Coverage
- **100% code coverage** for `readstring_context.go`
- 8 test functions covering all code paths and edge cases

## Test Cases
This PR implements tests for:
- ✅ Successful reads with various delimiters (newline, colon, space)
- ✅ Context cancellation scenarios (immediate and during read)
- ✅ Context timeout behavior
- ✅ Error handling (EOF, empty input)
- ✅ Edge cases (multiple delimiters, delimiter at end)
- ✅ Concurrent operations and context cancellation

## Test Results
All tests pass successfully:
```
PASS
ok      github.com/ansible/receptor/pkg/utils  0.024s
coverage: 100.0% for readstring_context.go
```

## Motivation
This addresses the need for better test coverage identified in the SonarCloud analysis. The `readstring_context.go` file had no test coverage previously, and these trivial-to-implement tests bring it to 100%.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/utils/readstring_context_test.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+	"errors"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ansible/receptor/pkg/utils"
+)
+
+func TestReadStringContext_SuccessfulRead(t *testing.T) {
```
Comment (fincamd) [inline pkg/utils/readstring_context_test.go]: I think we should settle with Camel Case as the recommended naming convention for functions in Go (or uppercase Camel Case for public functions). It could bring extra descriptive name to separate the name with underscores and I am also aware that we have a mixture of naming conventions, but this would bring us homogeneity in our function names

File: pkg/utils/readstring_context_test.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+			if got != tt.want {
+				t.Errorf("ReadStringContext() = %q, want %q", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestReadStringContext_ContextCancellation(t *testing.T) {
+	// Create a reader that will block (empty reader that never returns)
+	pr, pw := io.Pipe()
```
Comment (fincamd) [inline pkg/utils/readstring_context_test.go]: nitpicking here; could `readPipe` and `writePipe` be better names for these variables in all the tests? It adds mental load to try and decrypt every variable in the test case. Specially if someone new comes into the test cases and needs to make modifications or get info at a quick glance

**Review by fincamd — CHANGES_REQUESTED** [review]
Leaving some nitpick comments, otherwise LGTM. Thanks for the PR @PabloHiro

File: pkg/utils/readstring_context_test.go (hunk 1 above)
Comment (PabloHiro) [reply pkg/utils/readstring_context_test.go]: I am glad you took a deep look at this one, because I only validated that the tests made sense and coverage was bumped :pray:

File: pkg/utils/readstring_context_test.go [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ansible/receptor/pkg/utils"
+)
+
+func TestReadStringContextSuccessfulRead(t *testing.T) {
+	tests := []struct {
```
Comment (davemulford) [inline pkg/utils/readstring_context_test.go:18]: Suggest adding `t.Parallel()`

File: pkg/utils/readstring_context_test.go [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+			name:      "Read with space delimiter",
+			input:     "word1 word2",
+			delimiter: ' ',
+			want:      "word1 ",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := bufio.NewReader(strings.NewReader(tt.input))
```
Comment (davemulford) [inline pkg/utils/readstring_context_test.go:54]: Would need another `t.Parallel()` here.

File: pkg/utils/readstring_context_test.go [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+				t.Errorf("ReadStringContext() = %q, want %q", got, tt.wantStr)
+			}
+			if tt.wantErr && !errors.Is(err, io.EOF) {
+				t.Errorf("ReadStringContext() error = %v, want io.EOF", err)
+			}
+		})
+	}
+}
+
+func TestReadStringContextDelimiterAtEnd(t *testing.T) {
```
Comment (PabloHiro) [inline pkg/utils/readstring_context_test.go]: Remove this test, it is redundant

**Review by davemulford — CHANGES_REQUESTED** [review]
Looks good! Just the one comment about calling `t.Parallel()` in the tests.

File: pkg/utils/readstring_context_test.go (hunk 4 above)
Comment (PabloHiro) [reply pkg/utils/readstring_context_test.go:54]: Yup, it was compatible

File: pkg/utils/readstring_context_test.go (hunk 3 above)
Comment (PabloHiro) [reply pkg/utils/readstring_context_test.go:18]: Yup, it was compatible

**Review by fincamd — APPROVED** [review]

**Review by davemulford — APPROVED** [review]
LGTM! 💯
